numpy/core/arrayprint.py

```diff
@@ -632,7 +634,7 @@ def _formatArray(a, format_function, rank, max_line_len,
     if summary_insert and 2*edge_items < len(a):
         leading_items = edge_items
         trailing_items = edge_items
-        summary_insert1 = summary_insert + separator
```

Comment:
It felt a lot cleaner to me to do this concatenation here - can `lecacy` be forwarded to `_formatArray` instead?